pkg/authn/config.go

```diff
@@ -40,7 +40,9 @@ type AuthnConfig struct {
 
 // X509Config holds public client certificate used for authentication requests if specified
 type X509Config struct {
-	ClientCAFile string
+	ClientCAFile              string
+	UpstreamClientCertificate string
+	UpstreamClientKey         string
```

Comment:
`ClientCAFile` is in `authn.AuthnConfig` as it is used to authenticated downstream clients by their CN.

Wouldn't it be better to put it into the `main.tlsConfig`?

Reply:
indeed, that makes sense